toolkit/offchain/src/reserve/release.rs

```diff
@@ -106,7 +106,13 @@ fn reserve_release_tx(
 	let mut tx_builder = TransactionBuilder::new(&get_builder_config(ctx)?);
 
 	let reserve_balance = previous_reserve.utxo.get_asset_amount(&token);
-	let amount_to_transfer = cumulative_total_transfer - stats.token_total_amount_transferred;
+	let token_total_amount_transferred = stats.token_total_amount_transferred;
+	if cumulative_total_transfer <= token_total_amount_transferred {
+		return Err(anyhow!(
+			"The requested total amount: {cumulative_total_transfer} is not greater than the amount already transferred to illiquid supply: {token_total_amount_transferred}.",
+		));
+	}
+	let amount_to_transfer = cumulative_total_transfer - token_total_amount_transferred;
 	if amount_to_transfer as i128 > reserve_balance {
 		return Err(anyhow!("Not enough funds in the reserve to transfer {amount_to_transfer} tokens (reserve balance: {reserve_balance})"));
 	}
```

Comment:
I can't make suggestion for line 119, consider something like this:
```suggestion
	let reserve_balance: u64 = previous_reserve
		.utxo
		.get_asset_amount(&token)
		.try_into()
		.expect("Expected reserve_balance to be non-negative");

	let token_total_amount_transferred = stats.token_total_amount_transferred;
	let amount_to_transfer = cumulative_total_transfer.checked_sub(token_total_amount_transferred).ok_or_else(|| anyhow!("The requested total amount: {cumulative_total_transfer} is not greater than the amount already transferred to illiquid supply: {token_total_amount_transferred}."))?;

	let left_in_reserve = reserve_balance.checked_sub(amount_to_transfer).ok_or_else(|| anyhow!("Not enough funds in the reserve to transfer {amount_to_transfer} tokens (reserve balance: {reserve_balance})"))?;
```

Reply:
I've updated ogmios client and we don't need to `try_into()`